src/BenchmarksApps/Mvc/CustomJwtAuth.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using JsonWebToken;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+
+namespace Mvc
+{
+    public class CustomJwtAuth : AuthenticationHandler<AuthenticationSchemeOptions>
+    {
+        private readonly SymmetricJwk _key = new SymmetricJwk(Convert.FromBase64String("MFswDQYJKoZIhvcNAQEBBQADSgAwRwJAca32BtkpByiveJTwINuEerWBg2kac7sb"));
+        private JwtReader _reader;
```

Comment:
Same comment here, can these be reused across requests, are they thread-safe? Should be defined in the documentation. It's important for perf to reuse them if they are meant this way.

Reply:
Sure the key and the policy are safe to reuse, I kept the reader as instance since that is likely not thread safe